### PR TITLE
boot: simplify create_untypeds_for_region()

### PR DESCRIPTION
**Pull request by axel-h**
Simplify code and add comments.

### CODE REVIEW

File: src/kernel/boot.c [hunk 1]
```diff
@@ -601,27 +601,24 @@ BOOT_CODE static bool_t create_untypeds_for_region(
...
-        } else {
-            align_bits = size_bits;
-        }
-        /* Reduce size bits to align if needed */
-        if (align_bits < size_bits) {
-            size_bits = align_bits;
-        }
+        /* Calculate the bit size of the region. */
+        unsigned int size_bits = seL4_WordBits - 1 - clzl(reg.end - reg.start);
+        /* The bit size can't exceed a certain maximum. */
```
Comment (lsf37) [inline src/kernel/boot.c:608]:
```suggestion
        /* The size can't exceed the largest possible Untyped size. */
```

File: src/kernel/boot.c [hunk 2]
```diff
@@ -601,27 +601,24 @@ BOOT_CODE static bool_t create_untypeds_for_region(
...
+        /* The start address 0 satisfies any alignment needs, otherwise ensure
+         * the region's bit size does not exceed the alignment of the region.
+         */
+        if (0 != reg.start) {
+            unsigned int align_bits = ctzl(reg.start);
+            if (size_bits > align_bits) {
+                size_bits = align_bits;
+            }
+        }
+        /* Create a region only if it has a at least a certain size. */
```
Comment (lsf37) [inline src/kernel/boot.c:621]:
```suggestion
        /* Create a region only if it is large enough to be later retyped into objects. */
```

**Review by lsf37 — APPROVED** [review]
Would make the comments slightly more specific, otherwise I'd agree that this is slightly clearer.

**Comment by axel-h** [comment]
Thanks for the input, I've change the comments to be more clear.

**Pull request merged**